### ISSUE TITLE
S3 Auditor barfs when Watcher can't pull grantee

### ISSUE THREAD

**Issue by monkeysecurity** [Netflix/security_monkey#549]
https://github.com/Netflix/security_monkey/blob/develop/security_monkey/auditors/s3.py#L59

Problem:
Sometimes the S3 watcher retrieves `None` as the grantee on an s3 bucket instead of pulling the actual name.  When this occurs, the s3 Auditor barfs when attempting to call `.lower()` on `None`.

```
        acl = s3_item.config.get('Grants', {})
        for user in acl.keys():
            ...
            elif user.lower() in S3_ACCOUNT_NAMES:
                message = "ACL - Friendly Account Access."
                notes = "{} {}".format(",".join(acl[user]), user)
                self.add_issue(0, message, s3_item, notes=notes)
                ...
```

Short term solution:
Have the auditor `continue` when user is `None`.

Long term solution:
Fix the S3 auditor.

**Comment by alkyoox** [comment]
This seems to happen when s3 apis dont return the DisplayName in the grants section

**Comment by alkyoox** [comment]
as stated here: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGETacl.html
"This value will not be in the response in the US East (Ohio), Canada (Central), Asia Pacific (Mumbai), Asia Pacific (Seoul), EU (Frankfurt), EU (London), China (Beijing), or AWS GovCloud (US) regions."

**Comment by mikegrima** [comment]
@alkyoox : To appears to be the case for the `get_object_acl`, but for `get_bucket_acl`, that doesn't appear to be the case: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGETacl.html

**Comment by mikegrima** [comment]
This was fixed in #624.
